libs/community/langchain_community/llms/pieces.py

```diff
@@ -58,3 +50,33 @@ def _generate(
             text = self._call(prompt, stop=stop, run_manager=run_manager, **kwargs)
             generations.append([GenerationChunk(text=text)])
         return LLMResult(generations=generations)
+
+    def stream(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[GenerationChunk]:
+        """Stream the response from Pieces OS model."""
+        try:
+            for response in self.client.copilot.stream_question(prompt):
+                if response.question:
+                    answers = response.question.answers.iterable
+                    for answer in answers:
+                        yield GenerationChunk(text=answer.text)
+        except Exception as error:
+            print(f'Error streaming question: {error}')
+            yield GenerationChunk(text='Error streaming question')
+
+    def get_supported_models(self) -> List[str]:
```

Comment:
Well you should not use the enum instead you should use the /models endpoint

Reply:
We already have the implementation, you should do return self.client.available_models_names

Reply:
Addressed in [309eaff](https://github.com/shivay-at-pieces/langchain/pull/1/commits/309eaff69406bd38ae46b9cceb9c5a7cc2c094d7) commit